ArmoryQt.py

```diff
@@ -5230,21 +5267,35 @@ def unregisterProgressCallback(self, id):
    app_dir = "./"
    try:
       app_dir = os.path.dirname(os.path.realpath(__file__))
-   except:
-      if OS_WINDOWS and getattr(sys, 'frozen', False):
-         app_dir = os.path.dirname(sys.executable)
+   except Exception:
+      try:
+         # On frozen builds (Windows), sys.frozen is set to True
+         if OS_WINDOWS and sys.frozen:
+            app_dir = os.path.dirname(sys.executable)
+      except Exception:
+         pass
    translator.load(TheSettings.getGuiLanguage(), os.path.join(app_dir, "lang/"))
    QAPP.installTranslator(translator)
 
-   #setup main dialog
+   # 3) Create main window after setup manager is closed
    armoryMainWindow = ArmoryMainWindow(splashScreen=SPLASH)
 
-   #start cppbridge
-   TheBDM.startBridge(getBridgeArgList(), armoryMainWindow.networkReadyCallback)
+   # Bridge already started; ensure the main window receives the ready callback
+   try:
+      from armoryengine.CppBridge import TheBridge
+      if TheBridge.bridgeSocket.bip15xConnection.ready():
+         armoryMainWindow.networkReadyCallback()
+      else:
+         TheBridge.bridgeSocket.bip15xConnection.setNotifyReadyLbd(
+            lambda: QtCore.QMetaObject.invokeMethod(
+               armoryMainWindow, 'networkReadyCallback', QtCore.Qt.QueuedConnection))
```

Comment:
how? how can you get this far without the bridge running and setup manager handling the ready callback?